sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1024,26 +1033,29 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
   private[this] def castToIntervalCode(from: DataType): CastFunction = from match {
     case StringType =>
       (c, evPrim, evNull) =>
-        s"""$evPrim = CalendarInterval.fromString($c.toString());
+        code"""$evPrim = CalendarInterval.fromString($c.toString());
            if(${evPrim} == null) {
              ${evNull} = true;
            }
          """.stripMargin
 
   }
 
-  private[this] def decimalToTimestampCode(d: String): String =
-    s"($d.toBigDecimal().bigDecimal().multiply(new java.math.BigDecimal(1000000L))).longValue()"
-  private[this] def longToTimeStampCode(l: String): String = s"$l * 1000000L"
-  private[this] def timestampToIntegerCode(ts: String): String =
-    s"java.lang.Math.floor((double) $ts / 1000000L)"
-  private[this] def timestampToDoubleCode(ts: String): String = s"$ts / 1000000.0"
+  private[this] def decimalToTimestampCode(d: ExprValue): Block = {
+    val block = code"new java.math.BigDecimal(1000000L)"
```

Comment:
nit: why isn't this a literal?

Reply:
It looks like a statement to create `BigDecimal` object instead of just a `BigDecimal` literal?

Reply:
Does it matter? I think a literal is a piece of code (small) which is a left value and does not contain any reference to any reference to variables and can be put anywhere in the code returning always the same result. In the use case of splitting function, for instance, I imagine a literal is just reused everywhere needed without need to pass it to the new methods. So for these reasons IMHO this is a literal. WDYT? Also @cloud-fan WDYT?

Reply:
This piece of code that creates a `BigDecimal` object is simply used as an argument to `multiply` method. I think we may not need to track it as an `ExprValue`. Actually I also think here we can just `inline` it.

Reply:
maybe a `JavaCode.expression` then for this?

Reply:
`JavaCode.expression` is used to create `ExprValue` we need to track it in a code block. We don't need to track this `BigDecimal` variable. Here we just interpolate it into next code block.